src/dinit.cc

```diff
@@ -718,7 +723,19 @@ int dinit_main(int argc, char **argv)
             // Services all stopped but there was no shutdown issued. Inform user, wait for ack, and
             // re-start boot sequence.
             sync(); // Sync to minimise data loss if user elects to power off / hard reset
-            confirm_restart_boot();
+            if (auto_recovery) {
+                try {
+                    log_msg_end(" Will start recovery.");
+                    services->start_service("recovery");
+                }
+                catch (...) {
+                    std::cout << "Unable to start recovery service.\n";
+                    confirm_restart_boot();
+                }
+            } else {
```

Comment:
You don't really need the `else` clause, you can just have `confirm_restart_boot()` after the if.

I.e. rather than:

```
if (...) {
    try {
        ...
    }
    catch (...) {
        ...
        confirm_restart_boot();
    }
}
else {
    confirm_restart_boot();
}

```

You can have:

```
if (...) {
    try {
        ...
    }
    catch (...) {
        ...
    }
}
confirm_restart_boot();
```

The extra log message (" Will prompt user.") isn't really needed.

Reply:
The extra log thing was again to end https://github.com/davmac314/dinit/blob/927bec0/src/dinit.cc#L684 with a newline, but perhaps `shutdown_type_t::NONE` could be handled right below or be left as-is anyway, it's just currently missing a linebreak for the next thing appended to log

Reply:
I see, that actually makes sense (and I think in fact you're fixing an existing issue), though it's a bit hard to follow, so I might re-arrange things later. For now then this logic is good, however the `else` should be on the next line to match the code style (see CODE-STYLE).

Reply:
I've made it more clear what the intention was by using a more generic `Will handle boot failure.` where the others already were.